src/tests/ftest/checksum/csum_error_logging.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from daos_core_base import DaosCoreBase
+from dmg_utils import DmgCommand
+from avocado.utils import process
+
+
+class CSumErrorLog(DaosCoreBase):
+    """
+    Test Class Description: This test is runs
+    daos_test -z (Checksum tests) and verifies
+    whether Checksum Error Counters are incremented
+    in the NVME device due to checksum fault injection.
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-instance-attributes
+    def setUp(self):
+        super(CSumErrorLog, self).setUp()
+        self.dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        self.dmg.get_params(self)
+        self.dmg.hostlist.value = self.hostlist_servers[0]
+        self.dmg.set_sub_command("storage")
+        self.dmg.sub_command_class.set_sub_command("query")
+
+    def get_nvme_device_id(self):
+        self.dmg.sub_command_class.sub_command_class.set_sub_command("smd")
+        self.dmg.sub_command_class. \
+            sub_command_class.sub_command_class.devices.value = True
+        self.dmg.sub_command_class. \
+            sub_command_class.sub_command_class.pools.value = True
+        try:
+            result = self.dmg.run()
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+        uid = None
+        for line in result.stdout.splitlines():
+            line = line.strip()
+            if re.search("^UUID:",line):
+                uid = line.split()
+        return uid
+
+    def get_checksum_error_value(self, device_id=None):
+        if device_id is None:
+            self.fail("No device id provided")
+            return
+        self.dmg.sub_command_class. \
+            sub_command_class.set_sub_command("blobstore-health")
+        self.dmg.sub_command_class. \
+            sub_command_class. \
+            sub_command_class.devuuid.value = "{}".format(device_id)
+        try:
+            result = self.dmg.run()
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+        csum_count = None
+        for line in result.stdout.splitlines():
+            line = line.strip()
+            if re.search("^Checksum", line):
+                csum_count = line.split()
+        return csum_count
+
+    def test_csum_error_logging(self):
+        """
+        Test ID: DAOS-3927
+        Test Description: Write Avocado Test to verify single data after
+                          pool/container disconnect/reconnect.
+        :avocado: tags=all,full_regression,pr,hw,medium,csum_error_log
+        """
+        dev_id = self.get_nvme_device_id()
+        self.log.info("%s", dev_id)
+        csum = self.get_checksum_error_value(dev_id[1])
+        self.log.info("Checksum Errors : %s", csum[2])
+        DaosCoreBase.run_subtest(self)
+        csum_latest = self.get_checksum_error_value(dev_id[1])
+        self.log.info("Checksum Errors : %s", csum_latest[2])
+        if int(csum_latest[2]) > int(csum[2]):
```

Comment:
Instead of if/else, can we use self.assertTrue(csum_lastest>csum, "Checksum error log not incremented")

Reply:
sure. will change to assert.

Reply:
done.